client/src/api/tutorial/request.ts

```diff
@@ -189,8 +189,10 @@ export class TutorialAPI  {
 
   //MODAL HANDLERS
   public setModalVisible(data: any){
-    this.store.commit('SET_MODAL_VISIBLE',data);
-    this.completedActionWithImplicitForward();
+    if(data.data.activator != 'Server'){
```

Comment:
This conditional check makes me a bit squirmy. I'm not sure what the right design is since I don't know what `data.data` is, or what activator means, etc.

Reply:
Yeah, activator does mean that. With this implementation, a modal is generated every time an event comes in. This is fine in the game, and doesn't require any extra logic. In the Tutorial, we actually don't want that modal to appear, because it ruins the flow of everything else (we do the same thing with discard).

As for the data.data call, perhaps data:any should be renamed to modalInfo:ModalData, where ModalData is built of the modal types. (this may already be a thing @lfaivre, if you know, that would be great). Either way, the argument passed into the function has two attributes: type and data. Inside data, there is an activator attribute, which is just a string (Again, if @lfaivre hasn't done it already, we may look into defining a type for these values). 

This was just a quick way to prevent an undesired action in the Tutorial, but if you have any alternative suggestions, that would be great!